openvasd/openvasd.c

```diff
@@ -0,0 +1,2251 @@
+ */
+struct openvasd_connector
+{
+  char *ca_cert; /**< Path to the directory holding the CA certificate. */
+  char *cert;    /**< Client certificate. */
+  char *key;     /**< Client key. */
+  char *apikey;  /**< API key for authentication. */
+  char *server;  /**< original openvasd server URL. */
+  char *host;    /**< server hostname. */
+  char *scan_id; /**< Scan ID. */
+  int port;      /**< server port. */
+};
+
+/**
+ * @brief Struct holding options for Openvasd parameters.
+ */
+struct openvasd_param
+{
+  char *id;          /**< Parameter id. */
+  char *name;        /**< Parameter name. */
+  char *defval;      /**< Default value. */
+  char *description; /**< Description. */
+  char *type;        /**< Parameter type. */
+  int mandatory;     /**< If mandatory. */
+};
+
+/**
+ * @brief Struct credential information for Openvasd.
+ */
+struct openvasd_credential
+{
+  char *type;            /**< Credential type */
+  char *service;         /**< Service the credential is for */
+  char *port;            /**< Port the credential is for */
+  GHashTable *auth_data; /**< Authentication data (username, password, etc.)*/
+};
+
+/**
+ * @brief Struct holding target information.
+ */
+struct openvasd_target
+{
+  char *scan_id;            /**  Scan ID */
+  GSList *credentials;      /** Credentials to use in the scan */
+  char *exclude_hosts;      /** String defining one or many hosts to exclude */
+  char *hosts;              /** String defining one or many hosts to scan */
+  char *ports;              /** String defining the ports to scan */
+  char *finished_hosts;     /** String defining hosts to exclude as finished */
+  gboolean icmp;            /** Alive test method icmp */
+  gboolean tcp_syn;         /** Alive test method tcp_syn */
+  gboolean tcp_ack;         /** Alive test method tcp_ack */
+  gboolean arp;             /** Alive test method arp */
+  gboolean consider_alive;  /** Alive test method consider alive */
+  int reverse_lookup_unify; /** Value defining reverse_lookup_unify opt */
+  int reverse_lookup_only;  /** Value defining reverse_lookup_only opt */
+};
+
+/**
+ * @brief Struct holding vt information
+ */
+struct openvasd_vt_single
+{
+  char *vt_id;
+  GHashTable *vt_values;
+};
+
+/**
+ * @brief Request methods
+ */
+enum openvas_request_method
+{
+  POST,
+  GET,
+  HEAD,
+  DELETE,
+};
+
+typedef enum openvas_request_method openvasd_req_method_t;
+
+/** @brief Initialize a notus info struct and stores the server URL
+ *
+ *  @param server Original server to store and to get the info from
+ *
+ *  @return the initialized struct. NULL on error.
+ */
+openvasd_connector_t
+openvasd_connector_new (void)
+{
+  openvasd_connector_t connector;
+  connector = g_malloc0 (sizeof (struct openvasd_connector));
+  if (!connector)
+    return NULL;
+
+  return connector;
+}
+
+/** @brief Build a openvasd connector
+ *  @Description Recieve option name and value to build the openvasd connector
```

Comment:
Any reason you're using the description tag? Everywhere else has a blank line and then the description as text.

Also "receive".